crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -152,6 +152,18 @@ impl ProcessedShader {
                 Features::SHADER_PRIMITIVE_INDEX,
                 Capabilities::PRIMITIVE_INDEX,
             ),
+            (
+                Features::SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING,
```

Comment:
What happens when those features are not available? Should it just be user configurable instead of always asking for it?

Reply:
According to https://github.com/bevyengine/bevy/blob/49d50bd98011b1a4751896acdf368bc851c057af/crates/bevy_render/src/renderer/mod.rs#L136-L156 and https://github.com/bevyengine/bevy/blob/49d50bd98011b1a4751896acdf368bc851c057af/crates/bevy_render/src/settings.rs#L42-L79, on default, these features are only enabled when the adapter supports them, except being explicitly requested.

If those features are not available, not explicitly requested, and not used anyway as well, nothing bad would happen; If being not available while requested or used, there would be a validation error.